projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * Adds support for Jetpack Subscribe Modal feature
+ *
+ * @package automattic/jetpack-mu-wpcom
+ * @since $$next-version$$
+ */
+
+/**
+ * Jetpack_Subscribe_Modal class.
+ */
+class Jetpack_Subscribe_Modal {
+	/**
+	 * Jetpack_Subscribe_Modal singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Modal|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Modal instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Modal();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Jetpack_Subscribe_Modal class constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_loaded', array( $this, 'wpcom_create_subscribe_template' ) );
+		add_action( 'wp_footer', array( $this, 'wpcom_add_subscribe_modal_to_frontend' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'wpcom_enqueue_subscribe_modal_assets' ) );
+	}
+
+	/**
+	 * Adds a Subscribe Modal template part that can be edited in the editor.
+	 * This is later loaded in a pop up modal for Newsletter sites.
+	 *
+	 * @return void
+	 */
+	public function wpcom_create_subscribe_template() {
+		if ( $this->should_enable_subscriber_modal() ) {
+			$post = get_page_by_path( $this->get_subscribe_template_slug(), OBJECT, 'wp_template_part' );
+
+			if ( ! $post ) {
+				$template = array(
+					'slug'         => $this->get_subscribe_template_slug(),
+					'post_name'    => $this->get_subscribe_template_slug(),
+					'post_title'   => __( 'Subscribe Modal', 'jetpack' ),
+					'post_content' => $this->get_subscribe_template_content(),
+					'post_status'  => 'publish',
+					'post_author'  => 1,
+					'post_type'    => 'wp_template_part',
+					'scope'        => array(),
+					'tax_input'    => array(
+						'wp_theme' => get_option( 'stylesheet' ),
+					),
+				);
+				wp_insert_post( $template );
+			}
+		}
+	}
+
+	/**
+	 * Adds modal with Subscribe Modal content.
+	 *
+	 * @return void
+	 */
+	public function wpcom_add_subscribe_modal_to_frontend() {
+		if (
+			$this->should_enable_subscriber_modal() &&
+			! is_admin() &&
+			is_single() &&
+			'post' == get_post_type()
+		) {
+			$posts = get_posts(
+				array(
+					'post_type'   => 'wp_template_part',
+					'post_status' => 'publish',
+					'numberposts' => -1,
+				)
+			);
+
+			$subscribe_template = reset(
+				array_filter(
+					$posts,
+					function ( $post ) {
+						return $post->post_name === $this->get_subscribe_template_slug();
+					}
+				)
+			);
+
+			$blocks = parse_blocks( $subscribe_template->post_content );
```

Comment:
I think if we do `apply_filters('the_content', $post->post_content);` we can avoid doing the loop and `wp_kses`

Reply:
Hey @lezama - Couple questions on this one...

First...  Basically, we're taking some content from an editor template, content which is formatted in block format, and outputting outside the normal context, in a custom modal. My understanding is that you'd still need to correctly parse/convert the block markup to regular html mark up. Would the method you're describing somehow do that as well?

Second... I may be rusty on my filters, but apply_filter() I think just adds a new filter. Not sure how that would resolve the main issue just above. Did you mean to suggest add_filter()? Even in that case, I wouldn't think add_filter( 'the_content', $my_content ) wouldn't work here either. That filters the post content, but doesn't solve the issue of converting block markup to html markup.